crates/solidity/outputs/cargo/crate/src/backend/passes/p2_collect_definitions.rs

```diff
@@ -353,188 +335,85 @@ impl Visitor for Pass {
-                // constructor parameters scope.
-                if self.language_version < VERSION_0_5_0 {
+                } else if matches!(node.kind, input_ir::FunctionKind::Constructor) {
+                    // Register the constructor to resolve named parameters when
+                    // constructing this contract
                     self.register_constructor_parameters(parameters_scope_id);
                 }
-            } else {
-                self.insert_definition_in_current_scope(definition);
-            }
-        }
-
-        let function_scope =
-            Scope::new_function(node.node_id, self.current_scope_id(), parameters_scope_id);
-        let function_scope_id = self.enter_scope(function_scope);
-
-        if let Some(returns_declaration) = &node.returns {
-            self.collect_named_parameters_into_scope(
-                &returns_declaration.variables.parameters,
-                function_scope_id,
-            );
-        }
-
-        true
-    }
-
-    fn leave_function_definition(&mut self, node: &input_ir::FunctionDefinition) {
-        self.leave_scope_for_node_id(node.node_id);
-    }
-
-    fn enter_modifier_definition(&mut self, node: &input_ir::ModifierDefinition) -> bool {
-        let definition = Definition::new_modifier(node.node_id, &node.name);
-        self.insert_definition_in_current_scope(definition);
-
-        let modifier_scope = Scope::new_modifier(node.node_id, self.current_scope_id());
-        let modifier_scope_id = self.enter_scope(modifier_scope);
-        if let Some(parameters) = &node.parameters {
-            self.collect_named_parameters_into_scope(&parameters.parameters, modifier_scope_id);
-        }
-
-        true
-    }
-
-    fn leave_modifier_definition(&mut self, node: &input_ir::ModifierDefinition) {
-        self.leave_scope_for_node_id(node.node_id);
-    }
 
-    fn enter_constructor_definition(&mut self, node: &input_ir::ConstructorDefinition) -> bool {
-        let parameters_scope_id = self.collect_parameters(&node.parameters);
+                let function_scope =
+                    Scope::new_function(node.node_id, self.current_scope_id(), parameters_scope_id);
+                let function_scope_id = self.enter_scope(function_scope);
 
-        // Register the constructor to resolve named parameters when
-        // constructing this contract
-        self.register_constructor_parameters(parameters_scope_id);
-
-        let function_scope =
-            Scope::new_function(node.node_id, self.current_scope_id(), parameters_scope_id);
-        self.enter_scope(function_scope);
+                if let Some(returns) = &node.returns {
+                    self.collect_named_parameters_into_scope(returns, function_scope_id);
+                }
+            }
 
-        true
-    }
+            input_ir::FunctionKind::Modifier => {
+                let Some(name) = &node.name else {
+                    unreachable!("expected a name for the modifier");
```

Comment:
General question for a lot of these `unreachable`/`panic`/etc...

Is it possible they can be triggered by invalid user input? as the binder is still expected to run successfully/produce partial results...
Are there cases where we should be skipping silently instead of panicking?